smartsim/_core/_cli/cli.py

```diff
@@ -28,70 +28,84 @@
+from smartsim._core._cli.dbcli import execute as dbcli_execute
+from smartsim._core._cli.build import configure_parser as build_parser, execute as build_execute
 
 class SmartCli:
-    def __init__(self) -> None:
+    def __init__(self, menu: t.List[MenuItemConfig]) -> None:
+        self.menu: t.Dict[str, MenuItemConfig] = {item.command: item for item in menu}
         parser = argparse.ArgumentParser(
-            description="SmartSim command line interface", usage=_usage()
+            prog="smart",
+            description="SmartSim command line interface",
         )
-
-        parser.add_argument("command", help="Subcommand to run")
-
-        # smart
-        if len(sys.argv) < 2:
-            parser.print_help()
-            sys.exit(0)
-
-        args = parser.parse_args(sys.argv[1:2])
-        if not hasattr(self, args.command):
-            parser.print_help()
-            sys.exit(0)
-        getattr(self, args.command)()
-
-    def build(self) -> None:
-        Build()
-        sys.exit(0)
-
-    def clean(self) -> None:
-        Clean()
-        sys.exit(0)
-
-    def clobber(self) -> None:
-        Clean(clean_all=True)
-        sys.exit(0)
-
-    def site(self) -> None:
-        print(get_install_path())
-        sys.exit(0)
-
-    def dbcli(self) -> None:
-        bin_path = get_install_path() / "_core" / "bin"
-        for option in bin_path.iterdir():
-            if option.name in ("redis-cli", "keydb-cli"):
-                print(option)
-                sys.exit(0)
-        print("Database (Redis or KeyDB) dependencies not found")
-        sys.exit(1)
-
-
-def main() -> None:
-    SmartCli()
+        self.parser = parser
+        self.args: t.Optional[argparse.Namespace] = None
+
+        subparsers = parser.add_subparsers(dest="command",
+                                           required=True,
+                                           metavar="<command>",
+                                           help="Available commands")
+
+        for cmd, item in self.menu.items():
+            p = subparsers.add_parser(cmd, description=item.help, help=item.help)
+            if item.configurator:
+                item.configurator(p)
+
+    def execute(self, cli_args: t.List[str]) -> int:
+        if len(cli_args) < 2:
+            self.parser.print_help()
+            return 0
+
+        app_args = cli_args[1:]
+        self.args = self.parser.parse_args(app_args)
+
+        if not (menu_item := self.menu.get(app_args[0], None)):
+            self.parser.print_help()
+            return 0
+
+        return menu_item.handler(self.args)
+
+    @property
+    def parsed_args(self) -> t.Optional[argparse.Namespace]:
+        return self.args
+
+
+def default_cli() -> SmartCli:
+    build = MenuItemConfig("build",
+                           "Build SmartSim dependencies (Redis, RedisAI, ML runtimes)",
+                           build_execute,
+                           build_parser)
+    
+    cleanx = MenuItemConfig("clean",
```

Comment:
Not wrong, just curious: Any reason for the name `cleanx`? Is it just to avoid the name conflict with the global `clean` namespace?

Reply:
100% meant to avoid that collision. Getting rid of unused import solved it so this is no longer the `kleenex` variable